### PR TITLE
Make GCP OAuth scopes configurable via pipeline options.

### LINKED ISSUE #23290: [Feature Request]: Add Google Drive scope to GCP Auth
### What would you like to happen?

The current Python SDK doesn't include the `https://www.googleapis.com/auth/drive` scope when sending the authentication request to Google Cloud. This means that it's not possible to access Google Drive backed tables in BigQuery.

Adding the scope does indeed provide the required access. I'm only interested in the Python SDK, but this change should be easy to apply on all other SDKs as well if there's interest in that.

I can work on this change and submit a pull request.

### Issue Priority

Priority: 2

### Issue Component

Component: sdk-py-core

### PR DESCRIPTION
**Pull request by lukecwik**
This allows users to limit scopes dependent on their pipeline.

fixes #23290

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @kennknowles @tvalentyn @ricardograca-scratch

File: sdks/python/apache_beam/internal/gcp/auth.py [hunk 1]
```diff
@@ -169,7 +161,7 @@ def _get_service_credentials(pipeline_options):
       return None
 
   @staticmethod
-  def _add_impersonation_credentials(credentials, pipeline_options):
+  def _add_impersonation_credentials(credentials, scopes, pipeline_options):
```
Comment (tvalentyn) [inline sdks/python/apache_beam/internal/gcp/auth.py:164]: is there a reason we add a param instead of retrieving from pipeline_options?

**Review by tvalentyn — APPROVED** [review]

**Comment by tvalentyn** [comment]
Thank you

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by ricardograca-scratch** [comment]
@lukecwik Can you explain how this works? I upgraded Beam to version to 2.43.0, added the following:
```
pipeline_args = [
  ...,
  '--gcp_oauth_scope',
  'https://www.googleapis.com/auth/drive'
]
pipeline_options = PipelineOptions(flags=pipeline_args)
pipeline = beam.Pipeline(options=pipeline_options)
```
But still get 
```
Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials.
```

**Comment by lukecwik** [comment]
It looks like the error your getting is from BigQuery trying to access Drive on your behalf and not from Dataflow trying to access Drive. How are you trying to get BigQuery to access Drive from Dataflow (is it via your own code or is it via cross language IO connector or ...)?

**Comment by ricardograca-scratch** [comment]
I'm not using Dataflow. This job is running as a cronjob in Kubernetes and the credentials come from a service account. I can confirm the service account has the required role to access the BigQuery table, and the underlying Google Drive file is shared with the service account email. This same setup used to work just fine with my patched version of Beam that added the needed Auth scope.

**Comment by tvalentyn** [comment]
note that when setting `--gcp_oauth_scope`  you need to list all the scopes, not just the additional drive scope.

**Comment by ricardograca-scratch** [comment]
According to [the official documentation](https://docs.python.org/3/library/argparse.html#action) that is not the case when using the `append` option. It should just append the passed in scope to the default list, or am I missing something?

**Comment by tvalentyn** [comment]
I stand corrected, you don't need to repeat the default options.

> I'll try to debug the issue to see what could be wrong.

I am not sure what is going on then. I would print the content of the defined scopes and A/B test both cases.

**Comment by tvalentyn** [comment]
I tried supplying the option on the command line and can see that it is being populated correctly, but don't have a readily-available pipeline that I can use to meaningfully test a different scope.

**Pull request merged**